src/Exporter/SurveyExporter.php

```diff
@@ -50,41 +51,55 @@ public function export(Survey $survey, string $format, bool $allowedOnly = true)
-                    'Accepte d\'être invité à adhérer' => (int) $dataSurvey->getAgreedToContactForJoin(),
-                    'Code postal' => $allowPersonalData ? $dataSurvey->getPostalCode() : null,
-                    'Tranche d\'age' => $allowPersonalData && $dataSurvey->getAgeRange() ? $this->translator->trans('survey.age_range.'.$dataSurvey->getAgeRange()) : null,
-                    'Genre' => $allowPersonalData && $dataSurvey->getGender() ? (GenderEnum::OTHER === $dataSurvey->getGender() ? $dataSurvey->getGenderOther() : $this->translator->trans('common.'.$dataSurvey->getGender())) : null,
-                    'Accepte que ses données soient traitées' => (int) $dataSurvey->getAgreedToTreatPersonalData(),
-                    'Profession' => $allowPersonalData && $dataSurvey->getProfession() ? $dataSurvey->getProfession() : null,
-                ];
+                $allowPersonalData = $fromAdmin || $dataSurvey->getAgreedToTreatPersonalData();
+
+                $row = [];
+                $row['ID'] = ++$this->i;
+
+                if ($fromAdmin) {
+                    $row['Adherent ID'] = $dataSurvey->getAuthor()->getId();
+                }
+
+                $row['Nom Prénom de l\'auteur'] = $dataSurvey->getAuthor();
+                $row['Posté le'] = $dataSurvey->getPostedAt()->format('d/m/Y H:i:s');
+                $row['Nom'] = $allowPersonalData ? $dataSurvey->getFirstName() : null;
+                $row['Prénom'] = $allowPersonalData ? $dataSurvey->getLastName() : null;
+
+                if ($fromAdmin) {
+                    $row['Email'] = $dataSurvey->getEmailAddress() ;
+                    $row['Accepte d\'être contacté'] = (int) $dataSurvey->getAgreedToStayInContact();
+                    $row['Accepte d\'être invité à adhérer'] = (int) $dataSurvey->getAgreedToContactForJoin();
+                }
+
+                $row['Code postal'] = $allowPersonalData ? $dataSurvey->getPostalCode() : null;
+                $row['Tranche d\'age'] = $allowPersonalData && $dataSurvey->getAgeRange() ? $this->translator->trans('survey.age_range.'.$dataSurvey->getAgeRange()) : null;
+                $row['Genre'] = $allowPersonalData && $dataSurvey->getGender() ? (GenderEnum::OTHER === $dataSurvey->getGender() ? $dataSurvey->getGenderOther() : $this->translator->trans('common.'.$dataSurvey->getGender())) : null;
+
+                if ($fromAdmin) {
+                    $row['Accepte que ses données soient traitées'] = (int) $dataSurvey->getAgreedToTreatPersonalData();
+                }
+
+                $row['Profession'] = $allowPersonalData && $dataSurvey->getProfession() ? $dataSurvey->getProfession() : null;
 
                 /** @var SurveyQuestion $surveyQuestion */
                 foreach ($survey->getQuestions() as $surveyQuestion) {
+                    $questionName = $surveyQuestion->getQuestion()->getContent();
+                    $row[$surveyQuestion->getQuestion()->getContent()] = '';
```

Comment:
What about:
```suggestion
                    $row[$questionName] = '';
```
?

Reply:
yes good catch, I created a variable for that, and forgot to use it !